lua/entities/gmod_wire_expression2/core/files.lua

```diff
@@ -348,14 +348,14 @@ local function flushFileBufferInner()
 
 				if strlen > 0 then
 					net.Start("wire_expression2_file_download")
-					net.WriteUInt(2, 2)
-					net.WriteUInt(strlen, 32)
-					net.WriteData(data:sub(index + 1), strlen)
+						net.WriteUInt(2, 2)
+						net.WriteUInt(strlen, 32)
+						net.WriteData(data:sub(index + 1, index + 1 + strlen), strlen)
```

Comment:
Again, the substring is already limited by `WriteData` without needing to do extra addition.

Reply:
It's still currently making the substring function return a lot bigger of a string than it should which wastes cputime.